### ISSUE TITLE
Consider renaming vdlValue.String() to DebugString() and RawString() to String()

### ISSUE THREAD

**Issue by aghassemi** [vanadium/issues#866]
Currently in `vdlValue` the current naming pattern is `vdlValue.<Type>()`, for example `vdlValue.Byte()`, `vdlValue.Bool()`, etc.. based on that pattern, I expected `vdlValue.String()` to return the underlying string value but it is actually meant to return a human readable version of the value and `vdlValue.RawString()` is meant to be the one returning the underlying string value. 

To be consistent and to avoid confusion, consider we renaming the current `String()` to `DebugString()` and `RawString()` to `String()`.

(My misinterpretation of this caused https://github.com/vanadium/issues/issues/860)

@tatatodd @asadovsky

**Comment by tatatodd** [comment]
I agree that it's confusing, but the String method is special in Go:
  https://golang.org/pkg/fmt/#Stringer

So we really need vdl.Value.String to return a valid string representation for any vdl.Value.

That said, we could make the current vdl.Value.String do the same thing that it does today, except that it returns strings without quotes, and add a different vdl.Value.DebugString that adds the quotes.

**Comment by asadovsky** [comment]
That's a nice idea. If the implementation handles strings as the fast path, we wouldn't need "RawString" anymore, either. Do you think clients will care whether this method quotes strings? (How/where is it intended to be used? Is it intended to return strings that look like Go program literals where possible?)

**Comment by aghassemi** [comment]
Another option is to change the naming convention to something like `vdlValue.asByte()`, `vdlValue.asString()`, etc..
